app/models/enterprise.rb

```diff
@@ -71,36 +69,28 @@ class Enterprise < ActiveRecord::Base
   validate :name_is_unique
   validates :sells, presence: true, inclusion: {in: SELLS}
   validates :address, presence: true, associated: true
-  validates :email, presence: true
   validates_presence_of :owner
   validates :permalink, uniqueness: true, presence: true
   validate :shopfront_taxons
   validate :enforce_ownership_limit, if: lambda { owner_id_changed? && !owner_id.nil? }
   validates_length_of :description, :maximum => 255
 
-
-  before_save :confirmation_check, if: lambda { email_changed? }
-
   before_validation :initialize_permalink, if: lambda { permalink.nil? }
   before_validation :ensure_owner_is_manager, if: lambda { owner_id_changed? && !owner_id.nil? }
-  before_validation :ensure_email_set
   before_validation :set_unused_address_fields
   after_validation :geocode_address
 
   after_touch :touch_distributors
+  before_create :set_default_contact
   after_create :relate_to_owners_enterprises
-  # TODO: Later versions of devise have a dedicated after_confirmation callback, so use that
-  after_update :welcome_after_confirm, if: lambda { confirmation_token_changed? && confirmation_token.nil? }
-  after_create :send_welcome_email, if: lambda { email_is_known? }
+  after_create :send_welcome_email
 
   after_rollback :restore_permalink
 
 
   scope :by_name, order('name')
   scope :visible, where(visible: true)
-  scope :confirmed, where('confirmed_at IS NOT NULL')
-  scope :unconfirmed, where('confirmed_at IS NULL')
-  scope :activated, where("confirmed_at IS NOT NULL AND sells != 'unspecified'")
+  scope :activated, where("sells != 'unspecified'")
```

Comment:
Hm, possibly, but I don't think so. Trying to remember how the Devise confirmations logic goes. From memory: when an email address is changed, the new one is stored in the `unconfirmed_email` field until confirmation is completed, at which point it overwrites the value of `email` and `unconfirmed_email` is reset to nil. If you have a look in the old `EnterpriseConfirmationsController` you can see that we made it so that confirmations using an unknown email address automatically create a new `Spree::User` and then redirects to the reset password page so that they can set a password for themselves. So I am fairly confident that that means that any address in the `email` field on an enterprise should correspond to a user in the system...